main/src/ca/uwaterloo/flix/language/ast/TokenKind.scala

```diff
@@ -17,16 +17,332 @@ package ca.uwaterloo.flix.language.ast
+
+  case object ForceKeyword extends TokenKind
+
+  case object IfKeyword extends TokenKind
+
+  case object ImportKeyword extends TokenKind
+
+  case object InlineKeyword extends TokenKind
+
+  case object InstanceKeyword extends TokenKind
+
+  case object IntoKeyword extends TokenKind
+
+  case object LatKeyword extends TokenKind
+
+  case object LawKeyword extends TokenKind
+
+  case object LawfulKeyword extends TokenKind
+
+  case object LazyKeyword extends TokenKind
+
+  case object LetKeyword extends TokenKind
+
+  case object MatchKeyword extends TokenKind
+
+  case object TypeMatchKeyword extends TokenKind
+
+  case object NamespaceKeyword extends TokenKind
+
+  case object NullKeyword extends TokenKind
+
+  case object OpaqueKeyword extends TokenKind
+
+  case object OverrideKeyword extends TokenKind
+
+  case object ParKeyword extends TokenKind
+
+  case object YieldKeyword extends TokenKind
+
+  case object PubKeyword extends TokenKind
+
+  case object RefKeyword extends TokenKind
+
+  case object RegionKeyword extends TokenKind
+
+  case object ReifyKeyword extends TokenKind
+
+  case object ReifyBoolKeyword extends TokenKind
+
+  case object ReifyEffKeyword extends TokenKind
+
+  case object ReifyTypeKeyword extends TokenKind
+
+  case object RelKeyword extends TokenKind
+
+  case object SealedKeyword extends TokenKind
+
+  case object SetKeyword extends TokenKind
+
+  case object SpawnKeyword extends TokenKind
+
+  case object StaticKeyword extends TokenKind
+
+  case object TrueKeyword extends TokenKind
+
+  case object TypeKeyword extends TokenKind
+
+  case object UseKeyword extends TokenKind
+
+  case object WhereKeyword extends TokenKind
+
+  case object WithKeyword extends TokenKind
+
+  case object DiscardKeyword extends TokenKind
+
+  case object ObjectKeyword extends TokenKind
+
+  case object UppercaseName extends TokenKind
+
+  case object LowercaseName extends TokenKind
+
+  case object MathName extends TokenKind
+
+  case object GreekName extends TokenKind
+
+  case object Float32 extends TokenKind
+
+  case object Float64 extends TokenKind
+
+  case object Int8 extends TokenKind
+
+  case object Int16 extends TokenKind
+
+  case object Int32 extends TokenKind
+
+  case object Int64 extends TokenKind
+
+  case object BigInt extends TokenKind
+
+  case object BigDecimal extends TokenKind
+
+  case object String extends TokenKind
+
+  case object Char extends TokenKind
+
+  case object Decorator extends TokenKind // TODO: this refers to stuff like '@Internal'. Is this the right name for it?
+
+  case object AnonymousHole extends TokenKind
+
+  case object NamedHole extends TokenKind
+
+  case object VariableHole extends TokenKind
+
+  case object LineComment extends TokenKind
+
+  case object BlockComment extends TokenKind
+
+  case class Err(kind: LexerErr) extends TokenKind
 
   case object Eof extends TokenKind
 
-  case object Err extends TokenKind
+}
+
+sealed trait LexerErr
```

Comment:
Rework this to look like e.g. `WeederError` or `TypeError`.

Comment each error in detail, ideally with an example.

Reply:
OK, I realize that you probably need to keep these, because you use them in TokenKind.Err-- but we also need to connect them to some LexerError in the compiler.